core/domain/visualization_registry.py

```diff
@@ -51,16 +52,13 @@ def _refresh_registry(cls):
     @classmethod
     def get_full_html(cls):
         """Returns the HTML bodies for all visualizations."""
-        js_directives = (
-            utils.get_file_contents(os.path.join(
-                feconf.VISUALIZATIONS_DIR,
-                'OppiaVisualizationBarChartDirective.js')) +
-            utils.get_file_contents(os.path.join(
-                feconf.VISUALIZATIONS_DIR,
-                'OppiaVisualizationEnumeratedFrequencyTableDirective.js')) +
-            utils.get_file_contents(os.path.join(
-                feconf.VISUALIZATIONS_DIR,
-                'OppiaVisualizationFrequencyTableDirective.js')))
+        js_directives = ''
+        for filename in os.listdir(feconf.VISUALIZATIONS_DIR):
```

Comment:
Not listdir. Use get_all_visualization_ids() in this class, that's more robust. Then you shouldn't need to check for special filenames, the filename should already be there.

Reply:
done